### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by SmithChart**
This commit contains most of what I have learned while using flamingo.

It should give a small tour on the key-concepts of flamingo.

I am aware of some topics that are not covered by this. For example: plugins and the magic of the ``settings.py``. And I am quite sure that I am missing even more here.

@fscherf Feel free to update my README or even only pick a few changes.

Signed-off-by: Chris Fiege <c.fiege@pengutronix.de>

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -10,3 +10,116 @@ flamingo
...
+
+.. code-block: none
+
+   {% with entries=context.content.filter(type='blog').order_by('-date')[:5] %}
+   {% for entry in entries %}
+   {{ content.type }}: {{ content.url }}
+   {% endfor %}
+   {% endwith %}
+
+Executing flamingo
```
Comment (ejoerns) [inline README.rst]: Underline should match title length

File: README.rst [hunk 2]
```diff
@@ -10,3 +10,116 @@ flamingo
...
+----------------
+
+Flamingo can be executed in three different ways:
+
+* ``flamingo-build`` takes a given project and creates the static site.
+* ``flamingo-shell`` loads all content and invokes an iPython-shell where the
+  internal `context` is available.
+* ``flamingo-server`` can be used while creating content for a website.
+  In this mode flamingo builds the website and starts a webserver.
+  Every time a file in ``content`` is changed it is automatically rebuild.
```
Comment (ejoerns) [inline README.rst]: Maybe use `content/` or say '`content` directory' here.

File: README.rst [hunk 3]
```diff
@@ -10,3 +10,116 @@ flamingo
...
+
+content
+.......
+
+This directory contains the markup files with the actual content for the
+static site.
+Flamingo will scan all directories recursively.
+By default flamingo uses reStructuredText.
+
+Content files must follow a special format:
```
Comment (ejoerns) [inline README.rst]: Maybe explicitly note that it's rst with a special header

File: README.rst [hunk 4]
```diff
@@ -10,3 +10,116 @@ flamingo
...
+The first section of the file contains metadata of the file.
+The next section contains the actual markup.
+Both sections must be separated by a double newline.
+
+For every file in the ``content`` directory a ``Content`` object in the
+``context.contents`` -set is created.
+All metadata are represented as items of the ``Content`` object.
+The markdown section is represented in the special item ``content_body``.
+
+theme
```
Comment (ejoerns) [inline README.rst]: All directories but `output/` are described, thus maybe add a short sectio for `output/`, too?

File: README.rst [hunk 5]
```diff
@@ -10,3 +10,116 @@ flamingo
...
+   
+   
+   First headline
+   ==============
+   
+   The quick brown fox jumps over the lazy dog.
+
+The first section of the file contains metadata of the file.
+The next section contains the actual markup.
+Both sections must be separated by a double newline.
```
Comment (ejoerns) [inline README.rst]: Not sure if this restriction was removed, but if it exist, maybe we should emphasize this?

File: README.rst [hunk 6]
```diff
@@ -10,3 +10,116 @@ flamingo
 
 Flamingo is a python3-based, `pelican <https://blog.getpelican.com/>`_-inspired
 static site generator.
+
+
+Basic Concepts
+--------------
+
+Flamingo aims to convert content written in a markup language into a static
```
Comment (ejoerns) [inline README.rst]: `a markup language` or `reStructuredText`? I assumed flamingo is intentionally rst-only?

File: README.rst [hunk 7]
```diff
@@ -10,3 +10,116 @@ flamingo
...
+website. To archive this the flamingo core parses a directory of markup files
+and stores its content in an internal state called the ``context``. Plugins can
```
Comment (ejoerns) [inline README.rst]: Not sure if `context` should have code markup here. Isn't this here more referring to the 'concept' of a *context* than the actual potential code element? Maybe use `*context*` here instead.

**Comment by SmithChart** [comment]
@ejoerns thank you for your feedback. For most points I agree with you. If @fscherf is willing to take this PR (even if he removes the content after his documentation is finished) I will rebase my branch onto master and fix your suggestions.

**Comment by fscherf** [comment]
Hi @SmithChart and @ejoerns,

First of all: I am sorry it took me so long and thanks for writing docs!
Flamingo is a very young project, written from scratch with many new concepts. The reason why the most (and by most i mean almost all) features are underdocumented is i wasn't sure if they are going to last.

But i am almost done! v1.0 will come very soon!

@SmithChart 
The documentation in this pr is very specific to a particular flamingo project type but flamingo doesnt enforce to be used like this. In a few days i will release a command `flamingo init` which bootstraps flamingo projects.
I would ask you to port @ejoerns requested changes to the `README` of the first flamingo project template which will come with this feature.

A full api reference will come soon too.

**Comment by SmithChart** [comment]
> But i am almost done! v1.0 will come very soon!

A lot has changed in flamingo since I opened this PR. I will close it now.

**Pull request closed without merging**